File: bit_manipulation/1486_xor_operation_in_an_array/1486_xor_operation_in_an_array.py


class Solution:
    def xorOperation(self, n: int, start: int) -> int:
        ans = 0
        for i in range(n):
            ans ^= start + 2 * i
        return ans


n = 5
start = 0
res = Solution().xorOperation(n, start)
print(res)

File: bit_manipulation/1486_xor_operation_in_an_array/1486_xor_operation_in_an_array.go
package main

import "fmt"

func xorOperation(n int, start int) (xor int) {
	for i := 0; i < n; i++ {
		xor = xor ^ (start + 2*i)
	}
	return
}

func main() {
	n := 5
	start := 0
	res := xorOperation(n, start)
	fmt.Println(res)
}
